crypto/fipsmodule/rand/rand.c

```diff
@@ -291,33 +312,52 @@ static int rdrand(uint8_t *buf, size_t len) {
 
 #if defined(BORINGSSL_FIPS)
 
-static void CRYPTO_get_fips_seed(uint8_t *out_entropy, size_t out_entropy_len,
-                                 int *out_want_additional_input) {
-  *out_want_additional_input = 0;
-  // Every thread has its own Jitter instance so we fetch the one assigned
-  // to the current thread.
-  struct rand_thread_state *state =
-      CRYPTO_get_thread_local(OPENSSL_THREAD_LOCAL_RAND);
-  if (state == NULL) {
-    abort();
+#if defined(FIPS_ENTROPY_SOURCE_PASSIVE)
+void RAND_load_entropy(uint8_t out_entropy[CTR_DRBG_ENTROPY_LEN],
+                       uint8_t entropy[PASSIVE_ENTROPY_LOAD_LENGTH]) {
+
+  OPENSSL_memcpy(out_entropy, entropy, CTR_DRBG_ENTROPY_LEN);
+
+  // Whiten the resulting entropy
+  for (size_t i = 1; i < PASSIVE_ENTROPY_WHITEN_FACTOR; i++) {
+    for (size_t j = 0; j < CTR_DRBG_ENTROPY_LEN; j++) {
+      out_entropy[j] ^= entropy[CTR_DRBG_ENTROPY_LEN * i + j];
+    }
   }
```

Comment:
What does this do, do you have a link? Is there a reason to do it here and not in the Jitter/non-FIPS case?